### ISSUE TITLE
Device name clash with Alexa core commands

### ISSUE THREAD

**Issue by vaderag** [bwssytems/ha-bridge#753]
I have set up HA Bridge and all works well, however I am seeing an issue with naming which I am hoping someone can help with.

I have connected my Harmony Hub and set up a command for favorite channels - e.g. Cbeebies / BBC 1 to send the remote command to the hub and to the TV (e.g. 1, 0, 6, enter)

This works very occasionally, but usually Alexa returns "I have no media skills activated, please set up your media skill in the Alexa app..." or something along those lines.

I'm assuming it works when internet access is restricted or takes too long for Alexa to see the servers, but does anyone know a way to force your device to be the priority?

Thanks

**Comment by FloFoer** [comment]
How are the problematic devices exactly configured? (Names, Commands) 

Do i assume correct that you try to simulate the channel number input? When it doesn't work, does Alexa always say so or does she simply say "ok" sometimes and the desired action still doesn't happen? If it is the later it could also be a problem with your sequence, maybe you send the button presses to fast or something else.. 

> I'm assuming it works when internet access is restricted or takes too long for Alexa to see the servers, but does anyone know a way to force your device to be the priority?   

If there is a problem with the internet Alexa doesn't work at all, because all speech analysis (besides the wake word) is done online on amazon servers. Without a sufficient internet connection she doesn't know what you're saying.  
I don't know of a way to give your device priority, you just have to avoid ambiguous names AT ALL COST.  
Sometimes she does the stupid thing and asks "I found several devices matching that name. Which one did you mean?". A question you cant really answer with anything she'll understand. Or if the device name sounds just to similar to a known stock or skill action she simply tries that instead. 
 
If names are unique the only thing you have to have in mind is that the Echo has a timeout for command durations. I think it is 5 seconds. If your ha-bridge action takes longer than that she tries it a second time which is a bit stupid and if it still takes longer than 5 seconds she says something like "i don't know what went wrong" or "... isn't responding". This can especially be annoying for devices that act as a sort of scene. (I ran into problems with a device that does something like slowly dim ceiling light -> harmony activity -> slowly activate ambient lighting)

**Comment by vaderag** [comment]
Hi @FloFoer 

In the example I give, the name/description is: 
Cbeebies / Cbeebies, but I have also tried variants (C-Bee-bees / See Bee Bees) etc [nb. See Bee Bees actually work but you have to talk like you have a stutter else it picks up the core command.

When it doesnt work (90% of the time - funnily enough it's worked the few times I've tried it this morning) 

I noted the exact phrase she usually says tho - the full phrase she says is
"I couldnt find any enabled Video Skills. Go to the Music, Video and Books section of the Alexa app, enable the video provider you want  and then link your account and complete setup"

It seems ambiguity is the issue here because it clashes with a core command, it's a shame, but there isn't really a natural way around it, and the wife doesnt do unnattural commands ;)

**Comment by rodders99** [comment]
Media skills are not enabled. I found that I had to find names for media things that didn't conflict with reserved words and phrases that Alexa has. Switch on Netflix didn't work, switch movie time on, does work. You arenbattling reserved words, find new phrases e.g. change cbeebies to switch on kids TV perhaps ?

**Comment by vaderag** [comment]
@rodders99 yeah - think that's where I'm leaning towards - it's just that the wife already has the Xbox commands (Xbox Watch Cbeebies) in her head and I'm trying to steer her away from that so I can separate my Xbox and TV!
